modules/storages/lib/api/v3/storages/storage_representer.rb

```diff
@@ -94,6 +126,28 @@ def initialize(model, current_user:, embed_links: nil)
       { href: @connection_manager.get_authorization_uri, title: 'Authorize' }
     end
 
+    associated_resource :oauth_application,
+                        skip_render: ->(*) { !current_user.admin? },
+                        getter: ->(*) {
+                          ::API::V3::OAuth::OAuthApplicationsRepresenter.create(@oauth_application, current_user:)
+                        },
+                        link: ->(*) {
+                          {
+                            href: "/api/v3/oauth_applications/#{@oauth_application.id}",
+                            title: @oauth_application.name
+                          }
+                        }
+
+    associated_resource :oauth_client,
+                        as: :oauthClientCredentials,
+                        skip_render: ->(*) { !current_user.admin? || represented.oauth_client.blank? },
+                        representer: ::API::V3::OAuth::OAuthClientCredentialsRepresenter,
+                        link: ->(*) {
+                          return { href: nil } if represented.oauth_client.blank?
+
+                          { href: "/api/v3/oauth_client_credentials/#{represented.oauth_client.id}" }
```

Comment:
Why is overwriting the default link implementation necessary? Because of this, the link is always displayed even if the current_user is not an admin.

Reply:
overwrite was necessary:
- on oauth_apps: because we need this service result of the app and not the oauth app model fetched via active relation
- on oauth_client_credentials: because we have no title in this link

reason for apps might be only true for the getter, not the link. Can check that once I solved the other issues.

added apiv3 path helper for the href

Reply:
reworked representer code here